modules/inventory/server.lua

```diff
@@ -1342,6 +1342,37 @@ function Inventory.RemoveItem(inv, item, count, metadata, slot, ignoreTotal)
 end
 exports('RemoveItem', Inventory.RemoveItem)
 
+---@param inv inventory
+---@param items table
+---@param metadata? table | string
+function Inventory.CanCarryItems(inv, items, metadata)
+    inv = Inventory(inv) --[[@as OxInventory]]
+
+    local totalWeight = 0
+    for itemName, itemCount in pairs(totalItems) do
+        local item = Items(itemName)
+        if not item then goto skip end
+        
+        local itemSlots, _, emptySlots = Inventory.GetItemSlots(inv, item, type(metadata) == 'table' and metadata or { type = metadata or nil })
```

Comment:
Would be preferable to iterate over the inventory once, rather than per item. (requires new code)